src/proc/rotation-filter.cpp

```diff
@@ -52,32 +53,51 @@ namespace librealsense {
         register_option( RS2_OPTION_ROTATION, rotation_control ); 
     }
 
+    rotation_filter::rotation_filter( std::vector< rs2_stream > streams_to_rotate )
+        : stream_filter_processing_block( "Rotation Filter" )
+        , _streams_to_rotate( streams_to_rotate )
+        , _control_val( rotation_default_val )
+        , _real_width( 0 )
+        , _real_height( 0 )
+        , _rotated_width( 0 )
+        , _rotated_height( 0 )
+        , _value( 0 )
+    {
+        auto rotation_control = std::make_shared< ptr_option< int > >( rotation_min_val,
+                                                                       rotation_max_val,
+                                                                       rotation_step,
+                                                                       rotation_default_val,
+                                                                       &_control_val,
+                                                                       "Rotation angle" );
+
+        auto weak_rotation_control = std::weak_ptr< ptr_option< int > >( rotation_control );
+        rotation_control->on_set(
+            [this, weak_rotation_control]( float val )
+            {
+                auto strong_rotation_control = weak_rotation_control.lock();
+                if( ! strong_rotation_control )
+                    return;
+
+                std::lock_guard< std::mutex > lock( _mutex );
+
+                if( ! strong_rotation_control->is_valid( val ) )
+                    throw invalid_value_exception( rsutils::string::from()
+                                                   << "Unsupported rotation scale " << val << " is out of range." );
+
+                _value = val;
+            } );
+
+        register_option( RS2_OPTION_ROTATION, rotation_control );
+    }
+
     rs2::frame rotation_filter::process_frame(const rs2::frame_source& source, const rs2::frame& f)
     {
         if( _value == rotation_default_val )
```

Comment:
Should lock `_mutex` or place `_value` in a local variable and use it all over this function.

Reply:
Same for `update_output_profile` and `rotate_frame`.

Reply:
Done